### ISSUE TITLE
readOnly fields of a definition are not shown in example response of array type.

### ISSUE THREAD

**Issue by craffael** [swagger-api/swagger-ui#3762]
If we define a definitions object with readOnly properties and reference it from a response (that returns an array of objects), the readonly fields are not shown. 

| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  Bug
| Which Swagger/OpenAPI version?  | 2.0
| Which Swagger-UI version?       | 3.x
| How did you install Swagger-UI? | `docker run swaggerapi/swagger-ui:latest`
| Which browser & version?        | Chrome Version 61
| Which operating system?         | Ubuntu

### Demonstration API definition

```yaml
swagger: '2.0'
info:
  title: Test API
  version: 1.0.0

paths:
  /test:
    get:
      responses:
        200:
          description: success
          schema:
            type: array
            items:
              $ref: '#definitions/image'
definitions:
  image:
    type: object
    properties:
      url:
        type: string
        readOnly: true
      attributes:
        type: string
```

### Expected Behavior
The readonly field `url` should show up in swagger-ui under the example response.

### Current Behavior
The `url` field does not show up in the example response. (But it does in the model tab)
![screenshot from 2017-10-15 15-38-09](https://user-images.githubusercontent.com/15138672/31585236-ebe80090-b1be-11e7-8bf3-c6d61f8d1469.png)

### Context
I often forget to implement some properties of my object in the server side code because I rely on swagger-ui for api documentation.

**Comment by owenconti** [comment]
@craffael Can you please tell us your exact swagger-ui version? You can find it by running `window.versions.swaggerUi` in the console.

When I test your example spec against the latest (3.3.2 on petstore.swagger.io), I see the following:

<img width="1241" alt="screen shot 2017-10-15 at 8 47 53 am" src="https://user-images.githubusercontent.com/791222/31585827-9295b5e8-b185-11e7-879e-b2354a66e276.png">

http://petstore.swagger.io/?url=https://gist.githubusercontent.com/owenconti/a2a707de5a78873ccd268ee5ef24de47/raw/d01d6364518c4d938accf3dc286253c5d8358d08/%25233762#/default/get_test

**Comment by craffael** [comment]
Hmm I tried to run `window.versions.swaggerUi` in the console but this was not supported. I tried to figure out the actual version for a bit but ultimately just switched to the v3.3.2 docker tag (which should correspond to 3.3.2 git branch) and the problem was resolved.
It might have been caching issue in my browser, I'm a bit puzzled. But the issue is definitely resolved. Thanks for looking into it though!
